var/spack/repos/builtin/packages/coevp/package.py

```diff
@@ -0,0 +1,67 @@
+##############################################################################
+# Copyright (c) 2017, Los Alamos National Security, LLC
+# Produced at the Los Alamos National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Coevp(MakefilePackage):
+    """CoEVP is a scale-bridging proxy application for embedded viscoplasticity
+    applications. It is created and maintained by The Exascale Co-Design Center
+    for Materials in Extreme Environments (ExMatEx). The code is intended to
+    serve as a vehicle for co-design by allowing others to extend and/or
+    reimplement it as needed to test performance of new architectures,
+    programming models, etc.
+    Due to the size and complexity of the studied models, as well as
+    restrictions on distribution, the currently available LULESH proxy
+    application provides the coarse-scale model implementation and the ASPA
+    proxy application provides the adaptive sampling support."""
+
+    homepage = 'https://github.com/exmatex/CoEVP'
+
+    version('master', git='https://github.com/exmatex/CoEVP.git',
```

Comment:
call name it `develop`, see http://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

Reply:
Does that necessarily make sense in this case? CoEVP doesn't really follow a gitflow model and treats master as the end-all. And I notice spackages for projects with similar models (e.g. CoMD) also just refer to the "master" branch as "master"

Reply:
> Does that necessarily make sense in this case? 

yes

>  And I notice spackages for projects with similar models (e.g. CoMD) also just refer to the "master" branch as "master"

that's a mistake. The problem is, as soon as you have a release version, say, `x.y.z` and then somebody who use your package writes
```
depends_on(foo@x.y.z:)
```
then your `foo@master` aint gonna satisfy this dependency. So you are really advised to name the end-all as `develop`.

Reply:
p.s. there is a PR which makes `master` behave the same https://github.com/LLNL/spack/pull/1983 but it has not been merged.